### PR TITLE
Redundent cult convert checks

### PR DESCRIPTION
**Pull request by Aurorablade**
So uhh way back i had made it so you couldn't convert golems...or animals. (or i thought i did) I am hearing..that broke. I am fixing it now.

I know we got some new golem types but i am not getting runic golems in in this PR (and if anyone else wants to feel free), Mainly as i want a fix and runic golems need me to eval ease to make versus repercussion on round.

:cl:
fix:unbreaks letting golems be cult converts.
/:cl:

### CODE REVIEW

**Comment by Breenland** [comment]
Penguin cult no more? Bullshit!

Yea no that's fair. But could you imagine penguin cults...?

**Comment by datlo** [comment]
I'll implement runic golem specific to the cultists at some point, but for now it's good golems can't be culted. 👍

File: code/game/gamemodes/cult/runes.dm [hunk 1]
```diff
@@ -353,7 +353,7 @@ var/list/teleport_runes = list()
 	var/turf/T = get_turf(src)
 
 	for(var/mob/living/M in T.contents)
-		if(!iscultist(M) && !ismindshielded(M) && ishuman(M))
+		if(!iscultist(M) && !ismindshielded(M) && !isgolem(M) && !isanimal(M) && ishuman(M))
```
Comment (Citinited) [inline code/game/gamemodes/cult/runes.dm]: can't you just typecast the `for` loop to check for humans? Since nothing except humans will pass this check

Comment (Aurorablade) [reply code/game/gamemodes/cult/runes.dm]: Humans count for the mob not the speicies..and for some REASON golems are passing this check..as are penguins...

Comment (Citinited) [reply code/game/gamemodes/cult/runes.dm]: oh right. Still, typecasting would fix your penguin issue

Comment (Aurorablade) [reply code/game/gamemodes/cult/runes.dm]: all the is-calls (ishuman, isgolem) are ACTUALLY shorthand for a type check. This is pretty much me nuking from orbit.

**Comment by variableundefined** [comment]
I'm kinda reluctant about this PR because while I understand that golem will pass the check as a subtype of human...

I just have absolutely no idea how come penguin would pass the check? Is the report related to something else? Or is it another bug?

**Comment by Aurorablade** [comment]
The report is from an admin and is apprently recent. the only thing i can think of is that penguin is under /mob/living/simple_animal/pet/ so isanimal kinda checks for anything ANYTHING under /mob/living/simple_animal/ i think i actually had this trouble with mice before. why its letting penguins pass i dunno. But i am not gonna fight this. Updating PR in a sec.

**Comment by variableundefined** [comment]
I'm fine with adding the typecheck if there's no better solution and I get positive confirmation of it.

**Comment by Aurorablade** [comment]
you do realize that one if statment is nothing but typechecks anyway? but its all good.

**Pull request merged**